firmware/src/opaygo_functions.hpp

```diff
@@ -0,0 +1,313 @@
+
+int InvalidTokenCount = 0;
+
+void LoadActivationVariables() {
+  TokenCount =
+      mem.readInt(TokenCount_eeprom_location);  // We load TokenCount (&
+                                                // UsedTokens if needed)
+  UsedTokens =
+      mem.readInt(UsedTokens_eeprom_location);  // We load UsedTokens if needed
+  PAYGEnabled = mem.read(
+      PAYGEnabled_eeprom_location);  // We load PAYGEnabled //Verify this syntax
+  ActiveUntil =
+      mem.readLong(ActiveUntil_eeprom_location);  // We load ActiveUntil
+  TokenEntryLockedUntil = mem.readLong(
+      TokenEntryLockedUntil_eeprom_location);  // We load TokenEntryLockedUntil
+}
+
+void StoreActivationVariables() {
+  mem.writeInt(TokenCount_eeprom_location,
+               TokenCount);  // We store TokenCount (& UsedTokens if needed)
+  mem.writeInt(UsedTokens_eeprom_location,
+               UsedTokens);  // We store UsedTokens if needed
+  mem.write(PAYGEnabled_eeprom_location,
+            PAYGEnabled);  // We store PAYGEnabled //Verify this syntax
+  mem.writeLong(ActiveUntil_eeprom_location,
+                ActiveUntil);  // We store ActiveUntil
+  mem.writeLong(TokenEntryLockedUntil_eeprom_location,
+                TokenEntryLockedUntil);  // We store TokenEntryLockedUntil
+}
+
+void storeTimeStampEEPROM(uint32_t timeStampInSeconds) {
+  mem.writeInt(LAST_TIME_STAMP_ADDRESS, timeStampInSeconds);
+}
+
+void storeNbDisconnectionsEEPROM() {
+  mem.writeInt(NB_DISCONNECTIONS_ADDRESS, nbDisconnections);
+}
+
+void incrementNbDisconnectionsEeprom() {
+  Serial.println("Disconnection spotted!!");  // will be displayed if DEBUG_MODE
+                                              // is uncommented
+  nbDisconnections =
+      mem.readInt(NB_DISCONNECTIONS_ADDRESS);  // just to be sure we have the
+                                               // proper nb of disconnections
+  nbDisconnections++;
+  storeNbDisconnectionsEEPROM();
+}
+
+void ChangeLedState(int ledPin) {
+  digitalRead(ledPin) == LOW ? digitalWrite(ledPin, HIGH)
+                             : digitalWrite(ledPin, LOW);
+}
+
+void BlinkLED(int LedPin, int NumberOfBlinks, int BlinkPeriode) {
+  int i;
+  for (i = 0; i < NumberOfBlinks; i++) {
+    ChangeLedState(LedPin);
+    delay(BlinkPeriode);
+    ChangeLedState(LedPin);
+    delay(BlinkPeriode);
+  }
+}
+
+void BlinkRedLED(int NumberOfBlinks, int BlinkPeriode) {
+  BlinkLED(red_led, NumberOfBlinks, BlinkPeriode);
+}
+
+void BlinkGreenLED(int NumberOfBlinks, int BlinkPeriode) {
+  BlinkLED(green_led, NumberOfBlinks, BlinkPeriode);
+}
+
+uint32_t GetTimeInSeconds() {
+  /*
+   *Returns the unixtime in seconds of Now
+   */
+  DateTime now = rtc.now();
+  uint32_t nowInSeconds =
+      (now.unixtime() -
+       timeInitializationRtc);  // we substract the init time so that it's
+                                // easier to read when debugging, and works the
+                                // same as Arduino Time mgt
+  return (nowInSeconds);
+}
```

Comment:
This isn't PAYGO related, but time related, should it go in `time_management.hpp` instead?